src/Shared/TypeLoader.cs

```diff
@@ -192,10 +192,20 @@ private static Assembly LoadAssembly(AssemblyLoadInfo assemblyLoadInfo)
         private static Assembly LoadAssemblyUsingMetadataLoadContext(AssemblyLoadInfo assemblyLoadInfo)
         {
             string path = assemblyLoadInfo.AssemblyFile;
-            List<string> localPaths = new(Directory.GetFiles(Path.GetDirectoryName(path), "*.dll"));
-            localPaths.AddRange(runtimeAssemblies);
+            string[] localAssemblies = Directory.GetFiles(Path.GetDirectoryName(path), "*.dll");
 
-            _context = new(new PathAssemblyResolver(localPaths));
+            // Deduplicate between MSBuild assemblies and task dependencies. 
+            Dictionary<string, string> assembliesDictionary = new(localAssemblies.Length + runtimeAssemblies.Length);
```

Comment:
Bouncing through a dictionary here seems concerning, because it'll scramble the order. Is that always ok?

Reply:
Ah, looks like `PathAssemblyResolver`, surprisingly to me, uses a dictionary internally, so I guess it's ok.

Reply:
It should be fine anyway as long as it maintains the "only one" invariant. Ordering doesn't matter if there's only ever one valid match.